tests/components/rflink/test_init.py

```diff
@@ -380,3 +388,55 @@ async def test_not_connected(hass, monkeypatch):
     RflinkCommand.set_rflink_protocol(None)
     with pytest.raises(HomeAssistantError):
         await test_device._async_handle_command("turn_on")
+
+
+async def test_keepalive(hass, monkeypatch, caplog):
+    """Validate negative keepalive values."""
+    keepalive_value = -3
+    domain = RFLINK_DOMAIN
+    config = {
+        RFLINK_DOMAIN: {
+            CONF_HOST: "10.10.0.1",
+            CONF_PORT: 1234,
+            CONF_KEEPALIVE_IDLE: keepalive_value,
+        }
+    }
+
+    # setup mocking rflink module
+    _, _, _, _ = await mock_rflink(hass, config, domain, monkeypatch)
+    assert (
+        f"A bogus TCP Keepalive IDLE timer was provided ({keepalive_value} secs)"
+        in caplog.text
+    )
```

Comment:
Can we not only test that something was logged, but also add tests that the mocked `create_rflink_connection` was called with the appropriate value for each of these new tests? I gather `None` for this test and `test3_keepalive` below, and the given value for `test2_keepalive` despite the logged warning.